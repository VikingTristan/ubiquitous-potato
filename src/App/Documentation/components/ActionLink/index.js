import React from "react";

import { ComponentPreview, DocContainer, Property } from "@docutils";
import ActionLinkComponent from "@components/ActionLink";

const Overview = () => (
    <>
        <h2 id="overview">Overview</h2>
        <p>Action links.</p>
        <ComponentPreview language="html" showCasePanel codeFigure>
            <ActionLinkComponent linkText="My link"/>
        </ComponentPreview>
    </>
);

const badge = {
    type: "badge-default",
    text: "40"
};

const ActionLinkBadge = () => (
    <>
        <h2 id="action-link-badge">Action Link with Badge</h2>
        <p>Action links.</p>
        <ComponentPreview language="html" showCasePanel codeFigure>
            <ActionLinkComponent linkText="My link" badge={badge}/>
        </ComponentPreview>
    </>
);

const ActionLinkSmallText = () => (
    <>
        <h2 id="action-link-small-text">Action Link with small text</h2>
        <p>A small text can be displayed as a part of the Action Link. There are two types of Action Links with small text, one with the inline small text and one with the texts on different lines.</p>
        <ComponentPreview language="html" showCasePanel codeFigure>
            <ActionLinkComponent linkText="My link" smallText="This link directs to my page"/>
        </ComponentPreview>
        <p>Use <Property value=".action-link-multiline"/> for the Action Link with the texts on different lines.</p>
        <ComponentPreview language="html" showCasePanel codeFigure>
            <ActionLinkComponent linkText="My link" smallText="This link directs to my page" multiline={true} />
        </ComponentPreview>
    </>
);

const ActionLinkBadgeSmallText = () => (
    <>
        <h2 id="action-link-badge-small-text">Action Link with Badge and small text</h2>
        <p>Action Links can contain both Badge and small text</p>
        <ComponentPreview language="html" showCasePanel codeFigure>
            <ActionLinkComponent linkText="My link" smallText="This link directs to my page" badge={badge}/>
        </ComponentPreview>
        <ComponentPreview language="html" showCasePanel codeFigure>
            <ActionLinkComponent linkText="My link" smallText="This link directs to my page" multiline={true} badge={badge}/>
        </ComponentPreview>
    </>
);

const ActionLinkNewTab = () => (
    <>
        <h2 id="action-link-new-tab">Action Link that opens a new tab</h2>
        <p>An Action Link used in cases where a new tab is supposed to open is provided. Use <Property value=".action-link-new-tab" /> when it is known that the target will open in a new tab. </p>
        <ComponentPreview language="html" showCasePanel codeFigure>
            <ActionLinkComponent linkText="My link" newTab={true} />
        </ComponentPreview>
    </>
);

const ActionLink = () => (
    <DocContainer docToc>
        <p className="lead">
            Action link
        </p>
        <Overview />
        <ActionLinkBadge />
        <ActionLinkSmallText />
        <ActionLinkBadgeSmallText />
        <ActionLinkNewTab />
    </DocContainer>
);

export default ActionLink;

/* For testing */
export { Overview, ActionLinkBadge, ActionLinkSmallText, ActionLinkBadgeSmallText, ActionLinkNewTab };
