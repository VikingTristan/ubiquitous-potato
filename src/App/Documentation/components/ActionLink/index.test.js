import React from "react";
import { shallow } from "enzyme";

import ActionLink, { Overview, ActionLinkBadge, ActionLinkSmallText, ActionLinkBadgeSmallText, ActionLinkNewTab } from "./index";

describe("Components: ActionLink", () => {
    it("is defined", () => {
        expect(ActionLink).toBeDefined();
    });

    it("renders", () => {
        const wrapper = shallow(<ActionLink />);

        expect(wrapper).toMatchSnapshot();
    });

    describe("Overview", () => {
        it("is defined", () => {
            expect(Overview).toBeDefined();
        });

        it("renders", () => {
            const wrapper = shallow(<Overview />);

            expect(wrapper).toMatchSnapshot();
        });
    });

    describe("ActionLinkBadge", () => {
        it("is defined", () => {
            expect(ActionLinkBadge).toBeDefined();
        });

        it("renders", () => {
            const wrapper = shallow(<ActionLinkBadge />);

            expect(wrapper).toMatchSnapshot();
        });
    });

    describe("ActionLinkSmallText", () => {
        it("is defined", () => {
            expect(ActionLinkSmallText).toBeDefined();
        });

        it("renders", () => {
            const wrapper = shallow(<ActionLinkSmallText />);

            expect(wrapper).toMatchSnapshot();
        });
    });

    describe("ActionLinkBadgeSmallText", () => {
        it("is defined", () => {
            expect(ActionLinkBadgeSmallText).toBeDefined();
        });

        it("renders", () => {
            const wrapper = shallow(<ActionLinkBadgeSmallText />);

            expect(wrapper).toMatchSnapshot();
        });
    });

    describe("ActionLinkNewTab", () => {
        it("is defined", () => {
            expect(ActionLinkNewTab).toBeDefined();
        });

        it("renders", () => {
            const wrapper = shallow(<ActionLinkNewTab />);

            expect(wrapper).toMatchSnapshot();
        });
    });

});
