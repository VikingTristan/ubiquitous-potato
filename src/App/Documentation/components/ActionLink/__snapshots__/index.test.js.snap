// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Components: ActionLink ActionLinkBadge renders 1`] = `
<Fragment>
  <h2
    id="action-link-badge"
  >
    Action Link with Badge
  </h2>
  <p>
    Action links.
  </p>
  <ComponentPreview
    codeFigure={true}
    language="html"
    showCasePanel={true}
  >
    <ActionLink
      badge={
        Object {
          "text": "40",
          "type": "badge-default",
        }
      }
      linkText="My link"
    />
  </ComponentPreview>
</Fragment>
`;

exports[`Components: ActionLink ActionLinkBadgeSmallText renders 1`] = `
<Fragment>
  <h2
    id="action-link-badge-small-text"
  >
    Action Link with Badge and small text
  </h2>
  <p>
    Action Links can contain both Badge and small text
  </p>
  <ComponentPreview
    codeFigure={true}
    language="html"
    showCasePanel={true}
  >
    <ActionLink
      badge={
        Object {
          "text": "40",
          "type": "badge-default",
        }
      }
      linkText="My link"
      smallText="This link directs to my page"
    />
  </ComponentPreview>
  <ComponentPreview
    codeFigure={true}
    language="html"
    showCasePanel={true}
  >
    <ActionLink
      badge={
        Object {
          "text": "40",
          "type": "badge-default",
        }
      }
      linkText="My link"
      multiline={true}
      smallText="This link directs to my page"
    />
  </ComponentPreview>
</Fragment>
`;

exports[`Components: ActionLink ActionLinkNewTab renders 1`] = `
<Fragment>
  <h2
    id="action-link-new-tab"
  >
    Action Link that opens a new tab
  </h2>
  <p>
    An Action Link used in cases where a new tab is supposed to open is provided. Use 
    <Property
      value=".action-link-new-tab"
    />
     when it is known that the target will open in a new tab. 
  </p>
  <ComponentPreview
    codeFigure={true}
    language="html"
    showCasePanel={true}
  >
    <ActionLink
      linkText="My link"
      newTab={true}
    />
  </ComponentPreview>
</Fragment>
`;

exports[`Components: ActionLink ActionLinkSmallText renders 1`] = `
<Fragment>
  <h2
    id="action-link-small-text"
  >
    Action Link with small text
  </h2>
  <p>
    A small text can be displayed as a part of the Action Link. There are two types of Action Links with small text, one with the inline small text and one with the texts on different lines.
  </p>
  <ComponentPreview
    codeFigure={true}
    language="html"
    showCasePanel={true}
  >
    <ActionLink
      linkText="My link"
      smallText="This link directs to my page"
    />
  </ComponentPreview>
  <p>
    Use 
    <Property
      value=".action-link-multiline"
    />
     for the Action Link with the texts on different lines.
  </p>
  <ComponentPreview
    codeFigure={true}
    language="html"
    showCasePanel={true}
  >
    <ActionLink
      linkText="My link"
      multiline={true}
      smallText="This link directs to my page"
    />
  </ComponentPreview>
</Fragment>
`;

exports[`Components: ActionLink Overview renders 1`] = `
<Fragment>
  <h2
    id="overview"
  >
    Overview
  </h2>
  <p>
    Action links.
  </p>
  <ComponentPreview
    codeFigure={true}
    language="html"
    showCasePanel={true}
  >
    <ActionLink
      linkText="My link"
    />
  </ComponentPreview>
</Fragment>
`;

exports[`Components: ActionLink renders 1`] = `
<DocContainer
  docToc={true}
>
  <p
    className="lead"
  >
    Action link
  </p>
  <Overview />
  <ActionLinkBadge />
  <ActionLinkSmallText />
  <ActionLinkBadgeSmallText />
  <ActionLinkNewTab />
</DocContainer>
`;
